pandas/tests/dtypes/cast/test_infer_dtype.py

```diff
@@ -169,3 +170,14 @@ def test_cast_scalar_to_array(obj, dtype):
 
     arr = cast_scalar_to_array(shape, obj, dtype=dtype)
     tm.assert_numpy_array_equal(arr, exp)
+
+
+@pytest.mark.parametrize(
```

Comment:
its ok to test this i guess, but would need to parameterize on datetime64[D], ns, datetime64 etc and check the result

Reply:
done. I made all possible datetime formats except [ps] (as mentioned above)

Reply:
I think @jreback's comment was requesting testing those other frequencies here.

And then assert the result matches too.

```python
result = maybe_cast_to_datetime(obj, dtype)
expected = pd.Timestamp(...)  # right?
assert result == expected
```